### ISSUE TITLE
Machine ID: Add GitHub "Create Join Token" form

### ISSUE THREAD

**Issue by strideynet** [gravitational/teleport#44794]
Build on top of https://github.com/gravitational/teleport/pull/44408

Add GitHub join method support.

**Comment by zmb3** [comment]
Didn't @mcbattirola already build such a form?

**Comment by strideynet** [comment]
> Didn't @mcbattirola already build such a form?

Here I'm referring to the new Join Token page which supports creating/editing join tokens. We do have a discover flow which creates a GitHub join method token today, but it doesn't provide the ability to edit that token, and someone may want to create a token without the discover flow (e.g for a different usecase).

**Comment by zmb3** [comment]
Yep, I just want to make sure that we don't build two completely different forms for doing the same thing if someone picks this up.

**Comment by lukas8219** [comment]
@strideynet Is there any more specific requirements into this one?

**Comment by lukas8219** [comment]
@strideynet AFAIU from the previous PR - we already have the structure for the JoinTokenForm. We simply need to implement a Form and call the API to edit a resource like this [example](https://goteleport.com/docs/reference/join-methods/#github-actions-github) ?

**Comment by strideynet** [comment]
Closed by https://github.com/gravitational/teleport/pull/54308
